components/typeahead/typeahead.directive.ts

```diff
@@ -190,29 +151,36 @@ export class Typeahead implements OnInit {
     }
 
     if (this.typeaheadAsync === true) {
-      this.debouncer = this.debounce(() => {
-        if (typeof this.typeahead === 'function') {
-          this.typeahead().then((matches:any[]) => {
-            this._matches = [];
-            if (this.cd.model.toString().length >= this.typeaheadMinLength) {
-              for (let i = 0; i < matches.length; i++) {
-                this._matches.push(matches[i]);
-                if (this._matches.length > this.typeaheadOptionsLimit - 1) {
-                  break;
+      this.keyUpEventEmitter
+        .debounce(_ => Rx.Observable.timer(100))
+        .subscribe(
+          value => {
+            if (typeof this.typeahead === 'function') {
+              this.typeahead().then((matches:any[]) => {
+                this._matches = [];
```

Comment:
@valorkin this solution does not work (bug is possible?):

let a = new Array(1);
a.push('a');
a.push('b');
console.log(a);

out is: babel-node 1.ts 
[ , 'a', 'b' ]

in Angular2 result is same